### DISCUSSION TITLE
Group of figures and tables in twocolumn layout

### DISCUSSION THREAD

**Discussion by megane999** [quarto-dev/quarto-cli#4399] [Q&A] [answered]
How can i make a group of figures or|and tables in two column layout when render pdf?

Expected result
![Снимок экрана 2023-02-17 в 11 42 25](https://user-images.githubusercontent.com/2047240/219595882-852549e1-370b-48c3-9944-e2fad06832d0.png)
OR 
![Снимок экрана 2023-02-17 в 11 48 20](https://user-images.githubusercontent.com/2047240/219597214-1e0fc6e9-0230-4f1a-a5c4-fc0c9e886a96.png)

And how to avoid error "Package longtable Error: longtable not in 1-column mode." when I try to render twocolumn layout pdf with tables created with R Kable library?

**Comment by mcanouil** [comment]
Would you have a draft of something you tried?

You can include a small fake Quarto document following <https://quarto.org/bug-reports.html>.

**Reply by megane999** [reply]
> Would you have a draft of something you tried?
> 
> You can include a small fake Quarto document following https://quarto.org/bug-reports.html.

```
---
title: "Draft"
format:
  pdf:
    papersize: a4paper
    documentclass: article
    classoption: twocolumn
    keep-tex: true
    geometry:
      - showframe
      - top=20mm
      - left=20mm
editor: source
pdf-engine: lualatex
mainfont: "PT Serif"
---

## Header

Lorem ipsum dolor sit amet, consectetur adipiscing elit. In finibus sem quis sapien lacinia fermentum. Aliquam ut ornare ex. Mauris sed lectus hendrerit, eleifend dolor non, tempor felis. Vestibulum ac diam eget eros dignissim ultricies at quis nulla. Aliquam vitae lectus maximus nulla ullamcorper posuere vel non justo. In iaculis nibh id ex elementum tempus. Donec nec sapien ullamcorper, vestibulum nibh vel, semper nunc. Fusce venenatis maximus turpis non rutrum. Suspendisse bibendum erat eu nisi efficitur ultricies ut eget est. Aliquam faucibus erat nisl, sit amet ornare ligula elementum a. Quisque auctor dui sapien, vulputate placerat nibh dignissim sed. Pellentesque eleifend nulla sit amet est posuere convallis.

Aliquam tempus pellentesque ante, vel mattis metus convallis ac. Pellentesque mattis turpis ex, ut porttitor urna porta sed. Nullam varius iaculis urna et porttitor. Sed rhoncus mauris ex, dapibus semper ante facilisis vitae. Suspendisse tincidunt urna in ligula maximus mattis. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Quisque sit amet pellentesque eros. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec viverra sit amet mi sit amet imperdiet. Cras eget malesuada eros. Fusce tortor neque, semper ut enim quis, bibendum sollicitudin risus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Praesent at libero eu odio pellentesque consectetur. Morbi non urna nec enim convallis blandit.


## Header

Nulla rhoncus neque nec eros rhoncus, vitae pretium sem suscipit. Interdum et malesuada fames ac ante ipsum primis in faucibus. Aliquam vel risus id sem vestibulum luctus quis a sem. Donec sit amet velit quis elit imperdiet pulvinar. Donec dapibus ante quis urna luctus, ut sagittis sem rutrum. Etiam egestas, quam sit amet euismod scelerisque, urna nibh viverra libero, ut maximus mauris tellus id nisi. In rhoncus nec purus tempus scelerisque. Cras placerat egestas dapibus. Vestibulum ut rhoncus erat, sit amet suscipit dolor. Praesent vitae convallis lacus. Vestibulum eget ullamcorper risus.

Maecenas et libero nibh. Cras pretium convallis ornare. Nunc nec sapien pulvinar elit fermentum semper a eu purus. Pellentesque sollicitudin vehicula egestas. Vestibulum cursus gravida velit in condimentum. Maecenas et laoreet sapien. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam id odio dolor. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ac luctus velit, ut congue dui. Vestibulum non tortor quis nulla gravida feugiat. Pellentesque sollicitudin, nisl non iaculis cursus, urna eros ultrices massa, quis condimentum tellus nisl elementum diam. In venenatis sem at maximus gravida.

```{r}
#| label: fig1
#| fig-cap: "Cars"
#| warning: false
#| echo: false

plot(cars)
!```

```





tried every setting, but nothing help

```
#| column: body-right
#| column: screen-right

#| column: screen-inset-shaded
#| layout-nrow: 1

```

Found solution from tex.stackexchange

[link](https://tex.stackexchange.com/questions/34063/figure-span-to-one-column-on-double-column-page)

Don't know how to make same with quarto

**Comment by mcanouil** [comment, marked as answer]
Ok, so what you want is a figure that spread across columns.

Then, this is a LaTeX question, where the answer is to use the `figure*` environment and not `figure` (default).
`knitr` has the `fig.env` chunk option (see <https://yihui.org/knitr/options/#chunk-options>), which you can set `#| fig.env: figure*`.

**Reply by megane999** [reply]
Thank you very much!
It is working, but I i don't know what to do when I have option `#| layout-ncol: 2`

I mean rendered tex file don't change environment for two figures layout

Again:

```{r}
#| label: fig1
#| fig-cap: "Cars"
#| warning: false
#| echo: false
#| fig.env: figure*


plot(cars)
```

this code working good


![Снимок экрана 2023-02-22 в 13 06 48](https://user-images.githubusercontent.com/2047240/220588362-0285c85f-7992-46cf-83a3-e9631039b2e6.png)


```{r}

#| layout-ncol: 2
#| label: fig-cars2
#| fig-cap:
#|   - "Cars1"
#|   - "Cars2"
#| warning: false
#| echo: false
#| fig.env: figure*

plot(cars)

plot(pressure)

```

these one not working

![Снимок экрана 2023-02-22 в 13 34 14](https://user-images.githubusercontent.com/2047240/220595148-2c9fd568-4918-4a8e-b1fc-9297efb53c0a.png)




tex output

```
\begin{figure}

\begin{minipage}[t]{0.50\linewidth}

{\centering 

\raisebox{-\height}{

\includegraphics{quarto_files/figure-pdf/fig-cars2-1.pdf}

}

\caption{\label{fig-cars2-1}Cars1}

}

\end{minipage}%
%
\begin{minipage}[t]{0.50\linewidth}

{\centering 

\raisebox{-\height}{

\includegraphics{quarto_files/figure-pdf/fig-cars2-2.pdf}

}

\caption{\label{fig-cars2-2}Cars2}

}

\end{minipage}%

\end{figure}
```

**Reply by mcanouil** [reply]
What's the output you expect for the two plots?
To be as a single figure spreading across columns?
To have each of the subfigures to spread across columns?

**Reply by megane999** [reply]
To be a big panel with >2 Figures, and this panel should to be across all columns. 

example:
![Снимок экрана 2023-02-22 в 14 45 27](https://user-images.githubusercontent.com/2047240/220610923-ba58d716-8d55-41cd-aa47-63d6333401d8.png)

**Reply by mcanouil** [reply]
I suggest you use `patchwork` (<https://patchwork.data-imaginist.com/articles/guides/assembly.html>) which can handle both base and `ggplot2` plots.

Quarto layout is independent of computation, _i.e._, it can't combine plots like this.
Or in a different way, using subcaption, see <https://quarto.org/docs/authoring/figures.html#subcaptions>, but then, Quarto might use it's own LaTeX environment to set the minipage layout, this I would build one single figure/image using `patchwork` which would allow the annotation "A", "B", etc.

**Reply by megane999** [reply]
Thank you a lot, for your answer. I appreciate your feedback.

**Comment by ramiromagno** [comment]
Hi @mcanouil ,

I have the same question but in my case the image for the figure is not coming from the evaluation of a code chunk, i.e. I can't use `#| fig.env: figure*`. So is there a way of `![Elephant](elephant.png)`  generating a `\figure*{}`?

I think this might be quite common as many journal templates use two column layouts...

**Reply by mcanouil** [reply]
Why can't you?
The option is not specific to computation figures, see <https://quarto.org/docs/authoring/figures.html#latex-figures>.

**Reply by ramiromagno** [reply]
Because I am ignorant! :) I completely missed that from the documentation. Thanks a lot!
